win_redr_status(): skip if wildmenu is showing

justinmk opened the pull request:
This might be too coarse, but it passes all tests ...

A more nuanced approach might be: only skip the windows whose
statuslines are overwritten by the wildmenu.

Closes #2255
Closes #7108

- [x] mark `vim-path:8.0.0710` N/A

justinmk committed:
win_redr_status(): skip if wildmenu is showing

This might be too coarse, but it passes all tests ...

A more nuanced approach might be: only skip the windows whose
statuslines are overwritten by the wildmenu.

Closes #2255
Closes #7108

vim-patch:8.0.0710 N/A because of the changes in this commit.

justinmk commented on test/functional/ui/wildmode_spec.lua:
```diff
@@ -52,6 +54,59 @@ describe("'wildmode'", function()
+      screen:sleep(50)        -- Allow some output.
+      feed([[<C-\><C-N>gg]])
+      feed([[:sign <Tab>]])   -- Invoke wildmenu.
+      screen:sleep(50)        -- Allow some output.
```
The `screen:sleep()` here and in the next test are necessary to make sure some activity was allowed. But it also causes `screen:expect()` to wait until the full timeout. @bfredl I applied https://github.com/neovim/neovim/pull/6930 and found that it fixed that issue for both of these cases.

bfredl replied:
Hmm, but the "trivial success" detection logic in #6830 might flag it as a false positive. Rather we should allow to pass the min_timeout also.

jamessan commented on test/functional/ui/wildmode_spec.lua:
```diff
@@ -52,6 +57,59 @@ describe("'wildmode'", function()
+    end)
+
+    it('with laststatus=0, :vsplit, :term #2255', function()
+      command('set laststatus=0 shell=sh')
```
This should either be pending on Windows or have a Windows implementation, right (due to `shell=sh`)?

justinmk committed:
test/helpers: disable powershell logo/banner message

justinmk committed:
terminal: block redraw during c_CTRL-D

Unlike the normal wildmenu, the CTRL-D wild-list is not restored by
statusline redraw. (Semantics: ^D is controlled by 'wildoptions' option,
so it's in the "wild..." family.)

TODO: externalize the c_CTRL-D wild-list.

justinmk committed:
clint: allow starting brace after `enum`

justinmk committed:
build: silence maybe-uninitialized warning

False positive. From C:\msys64\mingw64\bin\gcc.exe (appveyor CI)

justinmk committed:
win_defs.h: redefine RGB macro after undefining

Before this change, if os_defs.h was included after macros.h then
win_defs.h would undefine our own RGB macro.

vim-patch:8.0.0146

justinmk committed:
input: skip dialogs if no UI is active

Treat dialogs in the same way as "silent mode" (`nvim -es`).

References #1984
References #3901
